packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -522,6 +570,61 @@ export default {
           }
           reactHooks.push(node.callee);
         }
+
+        // Check if a function created with useEvent is invoked within
+        // a useEffect or closure.
+        const scope = context.getScope();
+        const variable = recursivelyFindVariableInScope(
```

Comment:
Do we need to do this every time we see a call? Is there some way to mark event functions early (they occur only at top level and can be used only after they're defined) and then make this check very cheap?